core/src/main/java/hudson/Functions.java

```diff
@@ -1576,9 +1576,10 @@ public static boolean isAnonymous() {
     }
 
     /**
-     * Checks if the given authentication is anonymous, accordingly to {@link Jenkins#ANONYMOUS}
-     * @since TODO
+     * Checks if the given authentication is anonymous, accordingly to {@link Jenkins#ANONYMOUS}.
+     * TODO not the best place to put that method, but no other place at the moment
```

Comment:
I suggested `ACL`, perhaps.

Reply:
Yeah ACL was a good candidate but its responsability is more about authorization instead of authentication no ?

Reply:
Well, it has `SYSTEM` and `impersonate`/`as` as `static` utilities, and those are about authentication primarily. In fact it also has `ANONYMOUS` (not to be confused with `Jenkins.ANONYMOUS`, which is a different type) and some others.

Reply:
You convinced me 👍. 
So I also remove the `@Restricted` and use it in `Functions`.